### PR TITLE
Automate subscriber cancellation email 

### PR DESCRIPTION
**Pull request by sarvaiyanidhi**
This PR implements code to send mail on cancellation of subscription. 
More details in issue #795 

## Pull request checklist

- [x] My code contains tests covering the code I modified
- [x] I linted and tested the project with `bin/check`
- [x] I added significant changes and product updates to the [changelog](CHANGELOG.md)

Please note that dummy business hiring form route is added right now in routes.rb file which needs to be updated once business hiring form feature is merged into codebase.

### CODE REVIEW

File: app/views/business_mailer/cancel_subscription.text.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Hey <%= @business.first_name %>,
+
+Your subscription to RailsDevs is cancelled. Your access to the site will be available until the end of your billing period.
+
+Were you able to hire someone? <%= link_to "Fill out this form", business_hiring_form_url %> and I’ll send you info on next steps.
+
+A friendly reminder that according to the signed <%= link_to "hiring agreement", hiring_agreement_terms_url %> you must notify RailsDevs that you have hired someone no later than the developer's first day of employment.
```
Comment (joemasilotti) [inline app/views/business_mailer/cancel_subscription.text.erb:7]: We can't use HTML in plain text emails. Please update this to put the link at the end preceded with a colon. You can model it after `app/views/business_mailer/welcome.text.erb`.

File: app/views/business_mailer/cancel_subscription.text.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+<%= render "shared/signature" %>
```
Comment (joemasilotti) [inline app/views/business_mailer/cancel_subscription.text.erb:9]: I think you need to add an additional blank line to get rid of this warning.

File: app/views/business_mailer/cancel_subscription.html.erb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<p>Hey <%= @business.first_name %>,</p>
+
+<p>Your subscription to RailsDevs is cancelled. Your access to the site will be available until the end of your billing period.</p>
+
+<p>Were you able to hire someone? <%= link_to "Fill out this form", business_hiring_form_url %> and I’ll send you info on next steps.</p>
+
+<p>A friendly reminder that according to the signed <%= link_to "hiring agreement", hiring_agreement_terms_url %> you must notify RailsDevs that you have hired someone no later than the developer's first day of employment.</p>
+
+<%= render "shared/signature" %>
```
Comment (joemasilotti) [inline app/views/business_mailer/cancel_subscription.html.erb:9]: I think you need to add an additional blank line to get rid of this warning.

File: app/models/concerns/pay/subscription_extensions.rb [hunk 4]
```diff
@@ -5,6 +5,7 @@ module SubscriptionExtensions
     included do
       after_commit :send_admin_notification
       after_commit :send_subscribed
+      after_commit :send_cancel_subscription if Feature.enabled?(:cancel_subscription)
```
Comment (joemasilotti) [inline app/models/concerns/pay/subscription_extensions.rb:8]: This syntax looks weird. Shouldn't it be this?

```suggestion
      after_commit :send_cancel_subscription, if: Feature.enabled?(:cancel_subscription)
```

**Review by joemasilotti — CHANGES_REQUESTED** [review]
Looks great, thanks @sarvaiyanidhi! I made only a few code change requests.

File: app/views/business_mailer/cancel_subscription.text.erb (hunk 1 above)
Comment (sarvaiyanidhi) [reply app/views/business_mailer/cancel_subscription.text.erb]: Sorry did not realized while copying content from html file. Have fixed it.

File: app/views/business_mailer/cancel_subscription.text.erb (hunk 2 above)
Comment (sarvaiyanidhi) [reply app/views/business_mailer/cancel_subscription.text.erb]: Done. Added additional blank line after signature.

File: app/views/business_mailer/cancel_subscription.html.erb (hunk 3 above)
Comment (sarvaiyanidhi) [reply app/views/business_mailer/cancel_subscription.html.erb]: Done. Added additional blank line after signature.

File: app/models/concerns/pay/subscription_extensions.rb (hunk 4 above)
Comment (sarvaiyanidhi) [reply app/models/concerns/pay/subscription_extensions.rb]: Earlier code worked while testing but yeah it does not follow correct syntax. I have updated to use below syntax as the one mentioned in suggestion was throwing error. Have tested code locally.
     
after_commit :send_cancel_subscription, if: -> { Feature.enabled?(:cancel_subscription) }

Referred syntax from this article - https://engineering.gusto.com/the-rails-callbacks-best-practices-used-at-gusto/

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,7 @@
 
 ### March
 
+* March 9 - Automate subscriber cancellation email #795 @sarvaiyanidhi
```
Comment (joemasilotti) [inline CHANGELOG.md:7]:
```suggestion
* March 10 - Automate subscriber cancellation email #795 @sarvaiyanidhi
```

**Review by joemasilotti — APPROVED** [review]

**Comment by joemasilotti** [comment]
Thanks for this, @sarvaiyanidhi! Going to merge in when CI passes.

**Pull request merged**